src/EditorFeatures/Core/Implementation/IntelliSense/ModelComputation.cs

```diff
@@ -80,7 +81,9 @@ public Task<TModel> ModelTask
 
                 // We should never be called if we were stopped.
                 Contract.ThrowIfTrue(_stopCancellationToken.IsCancellationRequested);
+#pragma warning disable VSTHRD003 // Avoid awaiting foreign Tasks
                 return _lastTask;
+#pragma warning restore VSTHRD003 // Avoid awaiting foreign Tasks
```

Comment:
No `await` here so unclear why this is needed.

Reply:
The analysis is shallow. It assumes the context of a `Task` returned from an asynchronous method is the calling context at entry of the asynchronous method. If you return a different task, the analysis is invalidated.

Reply:
So basically this is going to flag us returning Tasks. That seems excessive.